src/ui/win32/bindings/MemoryViewerControlBinding.cpp

```diff
@@ -236,6 +239,68 @@ bool MemoryViewerControlBinding::HandleNavigation(UINT nChar)
     }
 }
 
+bool MemoryViewerControlBinding::HandleShortcut(UINT nChar)
+{
+    const bool bShiftHeld = (GetKeyState(VK_SHIFT) < 0);
+    const bool bControlHeld = (GetKeyState(VK_CONTROL) < 0);
+
+    switch (nChar)
+    {
+        case 'C':
+            if (bControlHeld)
+            {
+                const auto& pEmulatorContext = ra::services::ServiceLocator::Get<ra::data::context::EmulatorContext>();
+                const auto nValue = pEmulatorContext.ReadMemory(m_pViewModel.GetAddress(), m_pViewModel.GetSize());
+                std::wstring sValue = ra::data::MemSizeFormat(nValue, m_pViewModel.GetSize(), MemFormat::Hex);
+
+                ra::services::ServiceLocator::Get<ra::services::IClipboard>().SetText(ra::Widen(sValue));
+            }
+            return true;
+
+        case 'V':
+            if (bControlHeld and !m_pViewModel.IsReadOnly())
+            {
```

Comment:
This logic is sizable enough that I'd split it out into it's own function `OnPaste(bShiftHeld)`. Suggest doing the same for copy as well for consistency.